src/server/generic_family.cc

```diff
@@ -690,10 +691,12 @@ uint64_t ScanGeneric(uint64_t cursor, const ScanOpts& scan_opts, StringVec* keys
     };
 
     // Avoid deadlocking, if called from shard queue script
-    if (EngineShard::tlocal() && EngineShard::tlocal()->shard_id() == sid)
+    if (EngineShard::tlocal() && EngineShard::tlocal()->shard_id() == sid) {
       cb();
-    else
+      util::ThisFiber::Yield();
```

Comment:
Why do we yield on every iteration here ? We are also bound by `kMaxScanTimeMs` below ?

```
    if (time_now_ms > db_cntx.time_now_ms + kMaxScanTimeMs) {
      break;
    }
```

Reply:
While the scan call is limited by kMaxScanTimeMs, we want to not block the thread for this time therefore we have the timeout in OpScan

Reply:
we discussed internally :)